sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -679,6 +679,31 @@ void _nib_pl_remove(_nib_offl_entry_t *nib_offl)
 #endif  /* CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C */
 }
 
+void _nib_offl_remove_prefix(_nib_offl_entry_t *pfx)
+{
+    gnrc_netif_t *netif;
+
+    /* remove prefix timer */
+    evtimer_del(&_nib_evtimer, &pfx->pfx_timeout.event);
+
+    /* get interface associated with prefix */
+    netif = gnrc_netif_get_by_pid(_nib_onl_get_if(pfx->next_hop));
+
+    if (netif != NULL) {
+        /* remove address associated with prefix */
+        for (int i = 0; i < CONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF; i++) {
+            if (ipv6_addr_match_prefix(&netif->ipv6.addrs[i],
+                                       &pfx->pfx) >= pfx->pfx_len) {
+                gnrc_netif_ipv6_addr_remove_internal(netif,
+                                                     &netif->ipv6.addrs[i]);
+            }
+        }
```

Comment:
This should work better:
```suggestion
        uint8_t best_match_len = pfx->pfx_len;
        ipv6_addr_t *best_match = NULL;
        
        /* remove address associated with prefix */
        for (int i = 0; i < CONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF; i++) {
            if (ipv6_addr_match_prefix(&netif->ipv6.addrs[i],
                                       &pfx->pfx) >= best_match_len) {
                best_match_len = pfx->pfx_len;
                best_match = &netif->ipv6.addrs[i];
            }
        }
        if (best_match != NULL) {
            gnrc_netif_ipv6_addr_remove_internal(netif,
                                                 best_match);
        }
```

Reply:
yup, still works

Reply:
Not sure how this will improve things, if the match is > `pfx_len` that would be purely by chance if we assume the unused bits in a prefix to be undefined

Reply:
> Not sure how this will improve things, if the match is > `pfx_len` that would be purely by chance if we assume the unused bits in a prefix to be undefined

Not really sure what you mean :-/

Reply:
Settled offline. Just some confusion about addresses vs prefixes.